src/mmw/js/src/core/weatherStationLayer.js

```diff
@@ -5,30 +5,89 @@ var L = require('leaflet'),
     Marionette = require('../../shim/backbone.marionette'),
     weatherStationPopupTmpl = require('./templates/weatherStationPopup.html');
 
+var inactiveWeatherStationStyle = {
+    fill: true,
+    color: '#3388ff',
+    fillColor: 'steelblue',
+    fillOpacity: 0.2,
+    radius: 10,
+    weight: 3
+};
+
+var activeWeatherStationStyle = {
+    fill: true,
+    color: '#ff7800',
+    fillColor: '#ff7800',
+    fillOpacity: 0.5,
+    radius: 15,
+    weight: 5
+};
+
+var bindPopup = function(marker, feature, isActive) {
+    var view = new WeatherStationPopupView({ model: new Backbone.Model(feature.properties),
+                                             isActive: isActive });
+    marker.unbindPopup();
+    marker.bindPopup(view.render().el, { className: 'data-catalog-popover'});
+};
+
 var Layer = {
     createLayer: function(geojsonFeatureCollection) {
+        // If there is an active MapShed modeling project,
+        // we need to highlight the weather stations that were
+        // used in the model.
         return L.geoJson(geojsonFeatureCollection, {
             pointToLayer: function (feature, latlng) {
-                return L.circleMarker(latlng, {
-                    fill: true,
-                    fillColor: 'steelblue',
-                    fillOpacity: 0.2
-                });
+                return L.circleMarker(latlng, inactiveWeatherStationStyle);
             },
 
             onEachFeature: function(feature, marker) {
-                var model = new Backbone.Model(feature.properties),
-                    view = new WeatherStationPopupView({ model: model });
+                bindPopup(marker, feature, false);
+            }
+        });
+    },
 
-                marker.bindPopup(view.render().el, { className: 'data-catalog-popover'});
-            },
+    // Highlights the active weather stations in this layer, and returns
+    // a list of point geometries that represent the active weather stations.
+    setActiveWeatherStations: function(layer, activeWeatherStations) {
+        var activePoints = [];
+        layer.eachLayer(function(point) {
+            var props = point.feature.properties;
+            if(props.station in activeWeatherStations) {
+                activePoints.push(point.feature.geometry);
+                var ws = activeWeatherStations[props.station];
+                props.distanceToProjectAoi = (ws.distance / 1000.0).toFixed(2); // Use KM.
+                point.setStyle(activeWeatherStationStyle);
+                bindPopup(point, point.feature, true);
+            }
+        });
+        return activePoints;
+    },
+
+    clearActiveWeatherStations: function(layer, activeWeatherStations) {
+        layer.eachLayer(function(point) {
+            var props = point.feature.properties;
+            if(props.station in activeWeatherStations) {
```

Comment:
Seeing a failure on this line. To reproduce:

1. Select the City of Philadelphia-Schuylkill HUC-12
2. Go to MapShed
3. Turn on Weather Station layer
4. Click "New Project" to select another shape
5. See crash

![2020-02-26 17 00 14](https://user-images.githubusercontent.com/1430060/75392328-cd9c7b80-58b9-11ea-85fc-dc644e217b0c.gif)

Reply:
Not directly reproducible, seems to have been fixed in 94f3542.